### PR TITLE
Expandable Commit Header POC 3

### PR DESCRIPTION
**Pull request by tidy-dev**
### Screenshots

This is a screen shot of the app in it's smallest dimensions, but it should be noted that the commit list max is not to it's fullest, but everything is visible at that point as well. Zoom 200% on the other hand needs some more work.

https://github.com/desktop/desktop/assets/75402236/c15c34ac-e5fe-4bf7-8848-8bcfe125935e

## Description

POC 1: https://github.com/desktop/desktop/pull/17101
- Commit description below summary title
- Changed files count is on it's own row, Number of lines changed, and diff setting in their own  differentiated row by border and background color. 
- Commit long sha in front of short sha (expanded view)
- Çhanged files expands via popover

POC 2: https://github.com/desktop/desktop/pull/17100
- Commit description below summary title
- Changed files count is on it's own row, Number of lines changed, and diff setting in their own  differentiated row by border and background color. Similar to changes diff header.
- Çhanged files expands by appending drill down in expanded mode
- Commit long sha in front of short sha (expanded view)

POC 3: https://github.com/desktop/desktop/pull/17099
- Commit description below summary title
- Changed files, number of lines changed, and diff setting span the bottom row.
- Çhanged files expands by appending drill down in expanded mode
- Commit short sha to be in front of full sha. (expanded view)

POC 4: https://github.com/desktop/desktop/pull/17098
- Commit description below header.
- Commit short sha to be in front of full sha. (expanded view)
- Çhanged files expands by appending drill down in expanded mode

### Differences from original:
- Expand/Collapse in upper right hand corner
- Description is now beneath summary
- Diff settings right aligned -> similar to changes screen
- Sha displays a copy icon -> to be workshopped in to a button type thing, no longer has tooltip
- Changed file count says "+/- x changed files" if more than one type. If one type will say something like "[icon] x modified files."; no longer has tooltip
- The added/removed lines say "lines" after the count now. Tooltip removed.
- Tags is moved before modified lines in display -> thinking it having higher precedence than changed files and on expanded few the changed files count is closer to the diff/files list thus near related stuff.

### Expanded mode:
In short, the meta data switches from horizontal column display to vertical row display.
- The commit description expands inside a scrollable area differentiated by an alt box color background (feel kind of like a console/text area).
- Authors are listed vertically. with their name and email.
- commit sha display appends the full sha after short sha display -> maybe this is overkill since copy function will give user the long sha?
- tags no longer clipped with ellipsis, free to take up as many lines as needed.
- If there are multiple types of changed files, the changed files appends in parenthesis the drill down of types.
- The changed files, number of lines changed and diff settings span the bottom row.

### CODE REVIEW

**Comment by tidy-dev** [comment]
Just closing as will reopen when I have a version for review which will likely be a bit different.

We discussed and feedback is:
Main concern:
- Expand far left undiscoverable on a full maximized screen
-
Thoughts for improvement or Designer discussion:
- Diff option button moved to a diff header
- Files Changes header with number of kinds of files changed.
- Description below title or below meta data
- Add full/non-relative date to meta data (expanded view only?)
